### PR TITLE
Upgrade typescript-eslint-parser to 17.0.1

### PR DESCRIPTION
**Pull request by armano2**
typescript 2.9
typescript-eslint-parser: 17.0.1

solve #137

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,8 +34,8 @@
     "lint-staged": "^6.0.0",
     "mocha": "^4.0.1",
     "prettier": "^1.11.1",
-    "typescript": "~2.8.1",
-    "typescript-eslint-parser": "^15.0.0"
+    "typescript": "~2.9",
+    "typescript-eslint-parser": "^16.0.1"
```
Comment (macklinu) [inline package.json]: I checked this out locally to see how far we could increase the version of typescript + the parser before more tests started breaking, and it looks like we can upgrade to the following safely from this commit.

```
"typescript": "~3.0.3",
"typescript-eslint-parser": "^19.0.2"
```

Should we do that in this PR or a follow up? /cc @bradzacher

**Comment by armano2** [comment]
should i go forward with upgrading or wait for this to be merged?

**Comment by armano2** [comment]
typescript-eslint-parser v20 has some breaking changes, i will prepare fix for them

**Comment by bradzacher** [comment]
thanks @armano2!

feel free to branch off your branch so you can continue working whilst you wait for us to catch up.

in regards to the v20 compatibility.
we'll be releasing these changes as a breaking change to v1.0.0, so don't worry about backwards compatibility.

**Comment by armano2** [comment]
ok :>

**Pull request merged**